src/core/server/saved_objects/import/import_saved_objects.ts

```diff
@@ -69,6 +72,28 @@ export async function importSavedObjectsFromStream({
     supportedTypes,
     dataSourceId,
   });
+  // if not enable data_source, throw error early
+  if (!dataSourceEnabled) {
+    const notSupportedErrors: SavedObjectsImportError[] = collectSavedObjectsResult.collectedObjects.reduce(
+      (errors: SavedObjectsImportError[], obj) => {
+        if (obj.type === 'data-source' || isObjectWithDataSource(obj.id)) {
+          const error: SavedObjectsImportUnsupportedTypeError = { type: 'unsupported_type' };
+          const { title } = obj.attributes;
+          errors.push({ error, type: obj.type, id: obj.id, title, meta: { title } });
+        }
+        return errors; // Return the accumulator in each iteration
+      },
+      []
+    );
+    if (notSupportedErrors && notSupportedErrors.length > 0) {
```

Comment:
consider using optional chaining operator to simplify?
```suggestion
    if (notSupportedErrors?.length > 0) {
```